### PR TITLE
addition of discovery test

### PR DESCRIPTION
**Pull request by marinang**
Addition of the discovery test, plus addition of examples in docstrings of the discovery.py module and in the README.

A Jupyter notebook with a full example has been written, a binder badge will be added after this PR is merged in the master branch.

### CODE REVIEW

**Comment by marinang** [comment]
@eduardo-rodrigues  I have set up the pipelines using  Conda environments for windows, linux and Mac platforms. Now you can start reviewing when you can.

**Comment by eduardo-rodrigues** [comment]
OK, will do.

BTW, did you have a look at org scripts Henry provided in package https://github.com/scikit-hep/azure-wheel-helpers? They should make it eas-y/-ier to set things up consistently among packages, eventually … Maybe something to consider in the near future? I'm sure Henry will welcome feedback if you see things that are not exactly what's needed.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,56 @@
 # scikit-stats: statistics tools and utilities
 
+[![Build Status](https://dev.azure.com/matthieumarinangeli/matthieumarinangeli/_apis/build/status/scikit-hep.scikit-stats?branchName=master)](https://dev.azure.com/matthieumarinangeli/matthieumarinangeli/_build/latest?definitionId=3&branchName=master)
```
Comment (eduardo-rodrigues) [inline README.md]: At some point in the future we can use the scikit-hep account? No rush, of course, as having the CI via your account works just fine :-).

**Comment by marinang** [comment]
> OK, will do.
> 
> BTW, did you have a look at org scripts Henry provided in package https://github.com/scikit-hep/azure-wheel-helpers? They should make it eas-y/-ier to set things up consistently among packages, eventually … Maybe something to consider in the near future? I'm sure Henry will welcome feedback if you see things that are not exactly what's needed.

Yes I did have a look. What I have now is inspired by the those helpers, modified to work with Conda environments :).

File: README.md [hunk 2]
```diff
@@ -1,4 +1,56 @@
...
+>>> calculator = AsymptoticCalculator(loss, Minuit())
+>>> poinull = POI(Nsig, 0)
+>>> discovery_test = Discovery(calculator, [poinull])
+>>> discovery_test.result()
+
+p_value for the Null hypothesis = 0.0007571045424956679
+Significance = 3.1719464825102244
+```
+
+The discovery test prints out the pvalue and the significance of the null hypothesis to be rejected.
```
Comment (eduardo-rodrigues) [inline README.md]: Worth saying "significance (in units of sigma)" for complete clearness?

**Comment by eduardo-rodrigues** [comment]
> OK, will do.
> BTW, did you have a look at org scripts Henry provided in package https://github.com/scikit-hep/azure-wheel-helpers? They should make it eas-y/-ier to set things up consistently among packages, eventually … Maybe something to consider in the near future? I'm sure Henry will welcome feedback if you see things that are not exactly what's needed.
> 
> Yes I did have a look. What I have now is inspired by the those helpers, modified to work with Conda environments :).

Oh, nice! So there is scope for an extension of what's in there already … Again, something to put down for the future, maybe ...

File: skstats/hypotests/core/discovery.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                >>> loss = ExtendedUnbinnedNLL(model=[signal + background], data=[data], fit_range=[obs])
+
+                >>> from skstats.hypotests.calculators import AsymptoticCalculator
+                >>> from skstats.hypotests import Discovery
+                >>> from skstats.hypotests.parameters import POI
+
+                >>> calculator = AsymptoticCalculator(loss, MinuitMinimizer())
+                >>> poinull = POI(Nsig, 0)
+                >>> discovery_test = Discovery(calculator, [poinull])
+                >>> discovery_test.result()
```
Comment (eduardo-rodrigues) [inline skstats/hypotests/core/discovery.py]: I think this will fail doctests as there is no corresponding printout. Get back to it once doctests are running in the CI … (Sorry, I did not yet check whether they are running already :D.)

Comment (eduardo-rodrigues) [reply skstats/hypotests/core/discovery.py]: Ah! I try and have examples for anything that is not trivial, and then it really is important to have doctests running, to lower the burden of maintenance as much as possible. Indeed it is very easy to change something and quickly have examples in the code that no longer work :S. Having the doctests in `Particle` and `DecayLanguage` has been super useful IMHO.

File: README.md (hunk 1 above)
Comment (marinang) [reply README.md]: @eduardo-rodrigues agreed !
